internal/utils/extract.go

```diff
@@ -98,3 +98,29 @@ func stripComponents(headerName string, stripComponents int) string {
 	}
 	return filepath.Clean(strings.Join(names[stripComponents:], "/"))
 }
+
+func listHeaders(gzipStream io.Reader) ([]string, error) {
```

Comment:
Is this utility used somewhere else aside tests?
Otherwise i'd move it inside internal/utils/compress_test.go

Reply:
agreed, done!